### ISSUE TITLE
Dynamic Type should be supported throughout the app

### ISSUE THREAD

**Issue by bjtitus** [Automattic/pocket-casts-ios#1316]
### Description

When changing the preferred text size in iOS, Pocket Casts should resize text appropriately.

### Step-by-step reproduction instructions

1. Go to Settings > Accessibility > Display & Text Size > Larger Text
2. Change the slider at the bottom
3. Go to Pocket Casts
4. Check that text size has changed

### Did you search for existing bug reports?

#30 has a task related to this, but it is not strictly about text sizing.

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

iPhone 15 Pro Max, iOS 17.2, Pocket Casts 7.5.4

**Comment by bjtitus** [comment]
A few notes before I leave for AFK tomorrow.

I [began working on this](https://github.com/Automattic/pocket-casts-ios/tree/bjtitus/dynamic-type) for several screens and most of this can be done by enabling `adjustsFontForContentSizeCategory` and using `UIFont.TextStyle`s which are similar to our own. However, this will require going through every view and auditing every label, text view, and button so will take some time.

I think we have a choice here as to whether we want to reproduce our current font selection perfectly or whether we can vary some font sizes by a few points or change a few weights (as I did in my initial branch). [`UIFont+FontStyle`](https://github.com/Automattic/pocket-casts-ios/blob/a081912bba9b143a547c4023175f3d5e643c336f/podcasts/UIFont%2BFontStyle.swift) does have [methods](https://github.com/Automattic/pocket-casts-ios/blob/a081912bba9b143a547c4023175f3d5e643c336f/podcasts/UIFont%2BFontStyle.swift#L6) to make it a bit easier to tweak font weights with existing styles, but we will need to do that in code. Should be fine in most cases as most of these are already hooked up to `IBOutlets`. If we want to verify pixel perfect accuracy, we could add some kind of view snapshotting to ensure we aren't inadvertently changing things.

**Comment by emilylaguna** [comment]
I would love to have support for this. 

> However, this will require going through every view and auditing every label, text view, and button so will take some time.

This is the main concern I have, along with worrying about introducing new issues when making these changes. 

My thinking has been to add support where we can which has been on new views (written in SwiftUI) and views that we are updating. 

> [UIFont+FontStyle](https://github.com/Automattic/pocket-casts-ios/blob/a081912bba9b143a547c4023175f3d5e643c336f/podcasts/UIFont%2BFontStyle.swift) does have [methods](https://github.com/Automattic/pocket-casts-ios/blob/a081912bba9b143a547c4023175f3d5e643c336f/podcasts/UIFont%2BFontStyle.swift#L6) to make it a bit easier to tweak font weights with existing styles, but we will need to do that in code

The benefit of using UIFont+FontStyle is that we are able to specify the maximum font size that we want to support which gives us the trade off of being able to support larger font sizes, and allowing them to still look good within our designs.

**Comment by igotdes** [comment]
+1 in #7936880-zen:

> iOS has a feature to define app specific font sizes. Unfortunately it doesn‘t work in the list views of the app (newest episodes for example). Does it make something wrong or is it not implemented?

**Comment by joashrajin** [comment]
> Trying out Pocket Casts and liking what I’m seeing, however, there really needs to be a feature that allows for larger font/text size. Any chance of this feature coming to fruition? Or, am I overlooking the ability to do so? Thanks for the consideration!

https://forums.pocketcasts.com/forums/topic/ability-to-increase-font-text-ssze/?view=all

**Comment by joashrajin** [comment]
+1 at https://forums.pocketcasts.com/forums/topic/ability-to-increase-font-text-ssze/?view=all#post-3617

**Comment by joashrajin** [comment]
1 star review: 

> Dynamic Type Support has been requested for many years but they still haven't implemented it, or ANY way to increase text size.
It's crucial for accessibility but that doesn't seem to be a priority for them.

9128405-zd-a8c
